### ISSUE TITLE
winget list freezes

### ISSUE THREAD

**Issue by awg43** [microsoft/winget-cli#2104]
### Brief description of your issue

"winget list" freezes, although "winget search" works fine.

### Steps to reproduce

winget list
Results in a "freezed window" (the cursor on the title bar keeps running but no result is produced)

![winget list (window)](https://user-images.githubusercontent.com/9433327/163991293-d2dd1d59-c481-4c87-8d47-64e9e7e07fab.jpg)

=================
Maybe useful information:
===
PS C:\Users\Albert> winget --info
Windows Package Manager v1.2.10271
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22000.613
Package: Microsoft.DesktopAppInstaller v1.17.10271.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
====================================================
The (verbose) log result of the "winget list" command:
===
2022-04-19 12:47:03.926 [CORE] WinGet, version [1.2.10271], activity [{9B7A6F0D-3E1B-4F21-8996-BDE8D72387EF}]
2022-04-19 12:47:03.926 [CORE] OS: Windows.Desktop v10.0.22000.613
2022-04-19 12:47:03.926 [CORE] Command line Args: "C:\Users\Albert\AppData\Local\Microsoft\WindowsApps\winget.exe" list
2022-04-19 12:47:03.926 [CORE] Package: Microsoft.DesktopAppInstaller v1.17.10271.0
2022-04-19 12:47:03.926 [CORE] IsCOMCall:0; Caller: winget-cli
2022-04-19 12:47:03.931 [CLI ] WinGet invoked with arguments: 'list'
2022-04-19 12:47:03.931 [CLI ] Found subcommand: list
2022-04-19 12:47:03.931 [CLI ] Leaf command to execute: root:list
2022-04-19 12:47:03.931 [CLI ] Executing command: list
2022-04-19 12:47:03.933 [REPO] GetCurrentSourceRefs: Source named 'microsoft.builtin.desktop.frameworks' from origin Default is hidden and is dropped.
2022-04-19 12:47:03.933 [REPO] Default source requested, multiple sources available, adding all to source references.
2022-04-19 12:47:03.933 [REPO] Adding to source references msstore
2022-04-19 12:47:03.933 [REPO] Adding to source references winget
2022-04-19 12:47:03.934 [REPO] Source past auto update time [5 mins]; it has been at least 11 mins
2022-04-19 12:47:03.934 [REPO] Source past auto update time [5 mins]; it has been at least 11 mins
2022-04-19 12:47:05.057 [CORE] Examining extension: PFN = Microsoft.Winget.Source_8wekyb3d8bbwe, ID = IndexDB
2022-04-19 12:47:05.057 [CORE] Found matching extension.
2022-04-19 12:47:05.060 [REPO] Remote source data was not newer than existing, no update needed
2022-04-19 12:47:05.068 [REPO] Multiple sources available, creating aggregated source.
2022-04-19 12:47:05.068 [REPO] Adding to aggregated source: msstore
2022-04-19 12:47:05.068 [REPO] Sending http GET request to: https://storeedgefd.dsx.mp.microsoft.com/v9.0/information
2022-04-19 12:47:05.198 [REPO] Response status: 200
2022-04-19 12:47:05.199 [REPO] Sending http GET request to: https://storeedgefd.dsx.mp.microsoft.com/v9.0/information
2022-04-19 12:47:05.256 [REPO] Response status: 200
2022-04-19 12:47:05.257 [REPO] Adding to aggregated source: winget
2022-04-19 12:47:05.264 [CORE] Examining extension: PFN = Microsoft.Winget.Source_8wekyb3d8bbwe, ID = IndexDB
2022-04-19 12:47:05.264 [CORE] Found matching extension.
2022-04-19 12:47:05.287 [REPO] Opening SQLite Index for ImmutableRead at 'C:\Program Files\WindowsApps\Microsoft.Winget.Source_2022.419.1128.537_neutral__8wekyb3d8bbwe\Public\index.db'
2022-04-19 12:47:05.287 [SQL ] Opening SQLite connection: 'file:/C:/Program Files/WindowsApps/Microsoft.Winget.Source_2022.419.1128.537_neutral__8wekyb3d8bbwe/Public/index.db?immutable=1' [1, 40]
2022-04-19 12:47:05.288 [REPO] Opened SQLite Index with version [1.3], last write [2022-04-19 12:28:13.000]
2022-04-19 12:47:05.564 [REPO] Creating PredefinedInstalledSource with filter [None]
2022-04-19 12:47:05.564 [REPO] Creating new SQLite Index [4294967295.4294967295] at ':memory:'
2022-04-19 12:47:05.564 [SQL ] Opening SQLite connection: ':memory:' [6, 0]
2022-04-19 12:47:05.613 [REPO] Examining ARP entries for Machine | X64
2022-04-19 12:47:05.648 [REPO] Examining ARP entries for Machine | X86
2022-04-19 12:47:05.667 [FAIL] D:\a\_work\1\s\external\pkg\src\AppInstallerRepositoryCore\Microsoft\Schema\1_0\Interface_1_0.cpp(184)\WindowsPackageManager.dll!00007FF9C71F2465: (caller: 00007FF9C71EEC83) Exception(1) tid(1ab0) 800700B7 Kan geen bestand maken dat al bestaat.

2022-04-19 12:47:05.668 [REPO] Ignoring duplicate ARP entry Machine|X86|xampp [XAMPP]
2022-04-19 12:47:05.691 [REPO] Examining ARP entries for User | X64
=============================================
(By the way: i don't know why the preview shows some lines with in far larger font...)

### Expected behavior

I expected to get a list of programs that are available on my machine.
That list would be 20 or more lines (how many lines is just what i want to know) from the 3280+
lines shown by the command "winget search".

### Actual behavior

As described above.

### Environment

```shell
PS C:\Users\Albert> winget --info
Windows Package Manager v1.2.10271
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22000.613
Package: Microsoft.DesktopAppInstaller v1.17.10271.0
```

**Comment by lechacon** [comment]
@awg43 This is not related to your issue but
> (By the way: i don't know why the preview shows some lines with in far larger font...)

It's because of markdown. There are some lines that are just ===; these are marking the lines above as headers which show larger. You can prevent it by making all the logs show in `monospaced` font by surrounding them with three backticks ``` above and below.

**Comment by Trenly** [comment]
I've seen this behavior in Windows sandbox occasionally, where the thread will appear to freeze. I've found that pressing <kbd>Ctrl</kbd>+<kbd>C</kbd> can cause the task to resume.

**Comment by awg43** [comment]
I will keep the three backticks in mind. Thanks.
Pressing Ctrl+C does not help. Thanks.

**Comment by awg43** [comment]
To: denelon
Thanks for adding this issue to the issues which may be solved by next release.
Would any additional information help?

Do you have any suggestion on how/where to "dig in my PC", trying  to bypass this problem?

**Comment by denelon** [comment]
@awg43 
Milestones are aspirational, and I put all bugs in the next milestone. Most of them will not be completed in 1.3 and will be moved to 1.4.

We do have some theories on what is causing this issue. We're suspecting it may have to do with localizing the names of packages in Windows Apps & Features, but there may be other things going on.

Do you see the same problem if you search for a specific app?

For example: `winget list "App Installer"`

**Comment by awg43** [comment]
@denelon
First I updated to 7.2.3. Than used "winget reset source --force", than used "winget list". This resulted in
![winget source rest](https://user-images.githubusercontent.com/9433327/167208777-37003b06-b529-4c79-8521-3300e21cf716.jpg)
and again it freezes. Indeed this may caused by localization...

I also tested searching for a specific app (having the app "Everything" installed):
This resulted  also in a freezed output:
![winget list everything](https://user-images.githubusercontent.com/9433327/167209191-ff19145a-3ce6-478f-b70c-33bf00074c12.jpg)

**Comment by PeithonKing** [comment]
Did anyone succesfullly fix this issue? It still happens.

**Comment by Niessing-Manufaktur** [comment]
@PeithonKing I had this problem, where winget would freeze basically on any action, but I am not sure if the cause of it is the same as here.

For me, the problem was that I selected "world" as a region in my Windows settings. After I changed it to something more specific (Germany, in my example) I was able to update App Installer via Microsoft Store and then winget started to work again.

**Comment by Rev1k-Ej** [comment]
@PeithonKing I had the same problem. Not sure if the cause of it the same, but still.

For me, issue was that I had "World" selected as region in windows settings. After I changed it to something more specific (Germany, in my case), I was able to update "App-Installer" via MS Store and then Winget started to work again.

_For the record:
Paket: Microsoft.DesktopAppInstaller v1.21.2721.0_

**Comment by kcarvajalm** [comment]
i fixed with this 

![imagen](https://github.com/microsoft/winget-cli/assets/56368258/d2c5e031-69fd-4812-ad83-45afe9dc53d0)

**Comment by danielniccoli** [comment]
![grafik](https://github.com/microsoft/winget-cli/assets/2971735/ae4686d9-8147-4ebb-a930-f063c9697373)

```log
2024-01-06 20:17:37.118 [CORE] WinGet, version [1.2.10691], activity [{D4B89E35-5925-4132-A214-A999EF0FED3A}]
2024-01-06 20:17:37.118 [CORE] OS: Windows.Desktop v10.0.22631.2861
2024-01-06 20:17:37.118 [CORE] Command line Args: winget  source reset --force
2024-01-06 20:17:37.118 [CORE] Package: Microsoft.DesktopAppInstaller v1.17.10691.0
2024-01-06 20:17:37.118 [CORE] IsCOMCall:0; Caller: winget-cli
2024-01-06 20:17:37.121 [CLI ] WinGet invoked with arguments: 'source' 'reset' '--force'
2024-01-06 20:17:37.121 [CLI ] Found subcommand: source
2024-01-06 20:17:37.121 [CLI ] Found subcommand: reset
2024-01-06 20:17:37.121 [CLI ] Leaf command to execute: root:source:reset
2024-01-06 20:17:37.121 [CLI ] Executing command: reset
2024-01-06 20:17:37.129 [CLI ] Leaf command succeeded: root:source:reset
```

```log
2024-01-06 20:18:11.289 [CORE] WinGet, version [1.2.10691], activity [{A5940056-85A5-4B35-BC37-234334C75FC4}]
2024-01-06 20:18:11.289 [CORE] OS: Windows.Desktop v10.0.22631.2861
2024-01-06 20:18:11.289 [CORE] Command line Args: winget  install --id Microsoft.Powershell --source winget
2024-01-06 20:18:11.289 [CORE] Package: Microsoft.DesktopAppInstaller v1.17.10691.0
2024-01-06 20:18:11.289 [CORE] IsCOMCall:0; Caller: winget-cli
2024-01-06 20:18:11.294 [CLI ] WinGet invoked with arguments: 'install' '--id' 'Microsoft.Powershell' '--source' 'winget'
2024-01-06 20:18:11.294 [CLI ] Found subcommand: install
2024-01-06 20:18:11.294 [CLI ] Leaf command to execute: root:install
2024-01-06 20:18:11.297 [CLI ] Executing command: install
2024-01-06 20:18:11.302 [REPO] Named source requested, found: winget
2024-01-06 20:18:11.303 [REPO] Source past auto update time [5 mins]; it has been at least 28409478 mins

```

**Comment by Trenly** [comment]
> Same problem with a fresh install of Windows 11
> 
> ![grafik](https://private-user-images.githubusercontent.com/2971735/294712566-ae4686d9-8147-4ebb-a930-f063c9697373.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ1NzQyMDMsIm5iZiI6MTcwNDU3MzkwMywicGF0aCI6Ii8yOTcxNzM1LzI5NDcxMjU2Ni1hZTQ2ODZkOS04MTQ3LTRlYmItYTkzMC1mMDYzYzk2OTczNzMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDEwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAxMDZUMjA0NTAzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTBmYmQ5Y2UyZTM3ODI5YTRkOWRjNzZiOGMzZTBlZjNiYTA5ZjdmZDIzMGJjOGZhN2U2NTZmMzIwMWYxZGM4MiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.QLnYzXij70lBBP0yfege8cWLEQM6MIwhQLj6lTLlr2Y)
> 
> ```
> 2024-01-06 20:17:37.118 [CORE] WinGet, version [1.2.10691], activity [{D4B89E35-5925-4132-A214-A999EF0FED3A}]
> 2024-01-06 20:17:37.118 [CORE] OS: Windows.Desktop v10.0.22631.2861
> 2024-01-06 20:17:37.118 [CORE] Command line Args: winget  source reset --force
> 2024-01-06 20:17:37.118 [CORE] Package: Microsoft.DesktopAppInstaller v1.17.10691.0
> 2024-01-06 20:17:37.118 [CORE] IsCOMCall:0; Caller: winget-cli
> 2024-01-06 20:17:37.121 [CLI ] WinGet invoked with arguments: 'source' 'reset' '--force'
> 2024-01-06 20:17:37.121 [CLI ] Found subcommand: source
> 2024-01-06 20:17:37.121 [CLI ] Found subcommand: reset
> 2024-01-06 20:17:37.121 [CLI ] Leaf command to execute: root:source:reset
> 2024-01-06 20:17:37.121 [CLI ] Executing command: reset
> 2024-01-06 20:17:37.129 [CLI ] Leaf command succeeded: root:source:reset
> ```
> 
> ```
> 2024-01-06 20:18:11.289 [CORE] WinGet, version [1.2.10691], activity [{A5940056-85A5-4B35-BC37-234334C75FC4}]
> 2024-01-06 20:18:11.289 [CORE] OS: Windows.Desktop v10.0.22631.2861
> 2024-01-06 20:18:11.289 [CORE] Command line Args: winget  install --id Microsoft.Powershell --source winget
> 2024-01-06 20:18:11.289 [CORE] Package: Microsoft.DesktopAppInstaller v1.17.10691.0
> 2024-01-06 20:18:11.289 [CORE] IsCOMCall:0; Caller: winget-cli
> 2024-01-06 20:18:11.294 [CLI ] WinGet invoked with arguments: 'install' '--id' 'Microsoft.Powershell' '--source' 'winget'
> 2024-01-06 20:18:11.294 [CLI ] Found subcommand: install
> 2024-01-06 20:18:11.294 [CLI ] Leaf command to execute: root:install
> 2024-01-06 20:18:11.297 [CLI ] Executing command: install
> 2024-01-06 20:18:11.302 [REPO] Named source requested, found: winget
> 2024-01-06 20:18:11.303 [REPO] Source past auto update time [5 mins]; it has been at least 28409478 mins
> ```

You're using winget 1.2, and the CDN for that version has been deprecated. Try upgrading to 1.6 or 1.7-preview 

* #3652

**Comment by denelon** [comment]
We're on WinGet 1.7 now, and I believe this issue has been resolved.

**Comment by andreas-thoren** [comment]
This is still an issue maybe not with winget itself but If you install Win 11 through downloadable iso from microsoft sometime the region settings gets wrong. See this post @PeithonKing

"@PeithonKing I had the same problem. Not sure if the cause of it the same, but still.

For me, issue was that I had "World" selected as region in windows settings. After I changed it to something more specific (Germany, in my case), I was able to update "App-Installer" via MS Store and then Winget started to work again."

While the region setting is "World" Appinstaller does not get updated so you cannot fix the problem until region settings is fixed and you therefore can update Appinstaller.
